purchases/src/main/kotlin/com/revenuecat/purchases/errors.kt

```diff
@@ -0,0 +1,138 @@
+package com.revenuecat.purchases
+
+import com.android.billingclient.api.BillingClient
+import org.json.JSONException
+import java.io.IOException
+
+/**
+ * This class represents an error
+ * @param code Error code
+ * @param underlyingErrorMessage Optional error message with a more detailed explanation of the error  that originated this.
+ */
+class PurchasesError(
+    val code: PurchasesErrorCode,
+    val underlyingErrorMessage: String? = null
+) {
+    // Message explaining the error
+    val message: String = code.description
+
+    init {
+        errorLog("${code.description}${underlyingErrorMessage?.let { " | $it" }}")
+    }
+}
+
+enum class PurchasesErrorCode(val description: String) {
+    UnknownError("Unknown error."),
+    PurchaseCancelledError("Purchase was cancelled."),
+    StoreProblemError("There was a problem with the Play Store."),
+    PurchaseNotAllowedError("The device or user is not allowed to make the purchase."),
+    PurchaseInvalidError("One or more of the arguments provided are invalid."),
+    ProductNotAvailableForPurchaseError("The product is not available for purchase."),
+    ProductAlreadyPurchasedError("This product is already active for the user."),
```

Comment:
Perhaps we should not treat this as an error? On iOS 12.2 this is actually treated as a health case,

Reply:
Good idea, I am going to avoid return error and treat is a success, might help with some edge cases too

Reply:
Yeah, in that case it should process like a restore/successful purchase and return a purchaser info, ideally the latest. 

If that's too much of a refactor it can wait until 2.x+.

Reply:
I think I will merge and launch rc1 without it, and launch the stable with that change? I don't want to hold this more :)